Core.DomainServices/Queries/Helpers/ItContractIsActiveQueryHelper.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using Core.DomainModel.ItContract;
+
+namespace Core.DomainServices.Queries.Helpers
+{
+    internal class ItContractIsActiveQueryHelper
+    {
+        public static bool CheckIfContractIsValid(DateTime currentTime, ItContract contract)
```

Comment:
no no no... you are breaking the expression here... it's an expression tree which must be compilable into SQL.. that is killed by this now.